### PR TITLE
WIP: add flink service responses

### PR DESCRIPTION
**Pull request by mhoffm-aiven**

### CODE REVIEW

File: flink_job.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	// CreateFlinkJobResponse Aiven API response
+	// POST https://api.aiven.io/v1/project/<project>/service/<service_name>/flink/job
+	CreateFlinkJobResponse struct {
+		APIResponse
+
+		JobName string `json:"job_name"`
+		JobId   string `json:"job_id"`
+	}
+
+	// DeleteFlinkJobRequest Aiven API request
```
Comment (ivan-savciuc) [inline flink_job.go:26]: Probably it should be a `PatchFlinkJobRequest` here?

File: flink_job.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+			Running      int `json:"RUNNING"`
+			Scheduled    int `json:"SCHEDULED"`
+		} `json:"timestamps"`
+
+		//TODO: if this is needed, add fields
+		Vertices []interface{} `json:"vertices"`
+	}
+)
+
+// Creates a flink job
```
Comment (ivan-savciuc) [inline flink_job.go:94]: To make go doc happy for all the comments should have a prefix `{function name}`, can you add here //**Create** creates a Flink job?

File: flink_table.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+	// CreateFlinkTableResponse Aiven API response
+	// POST https://api.aiven.io/v1/project/<project>/service/<service_name>/flink/table
+	CreateFlinkTableResponse struct {
+		APIResponse
+
+		flinkTable
+	}
+
+	// GetFlinkTableResponse Aiven API request
```
Comment (ivan-savciuc) [inline flink_table.go:29]: Here we should probably use `GetFlinkTableRequest`

File: flink_table.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+	// GetFlinkTableResponse Aiven API response
+	// GET https://api.aiven.io/v1/project/<project>/service/<service_name>/flink/table/<table_id>
+	GetFlinkTableResponse struct {
+		APIResponse
+
+		flinkTable
+	}
+
+	// DeleteFlinkTableResponse Aiven API response
```
Comment (ivan-savciuc) [inline flink_table.go:43]: Should be a `DeleteFlinkTableRequest` here I think

File: flink_table.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	// DELETE https://api.aiven.io/v1/project/<project>/service/<service_name>/flink/table/<table_id>
+	DeleteFlinkTableRequest struct {
+		TableId string `json:"table_id"`
+	}
+
+	// ListFlinkTableRequest Aiven API request
```
Comment (ivan-savciuc) [inline flink_table.go:49]: And here `ListFlinkTableResponse`

File: flink_table.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+	// shared fields by some responses
+	flinkTable struct {
+		IntegrationId string `json:"integration_id"`
+		TableId       string `json:"table_id"`
+		TableName     string `json:"table_name"`
+	}
+)
+
+// Creates a flink table
```
Comment (ivan-savciuc) [inline flink_table.go:65]: Also here in the few functions bellow lets use a {function name} prefix for all the comments

**Review by ivan-savciuc — DISMISSED** [review]
Have some cosmetic suggestions, otherwise LGTM.

One thing which is beyond this PR, what should we do with the ids?  We have no consistency in the code in this regard, in some places we use `Id` in some `ID`. golangci linter expects it to be `ID`. At some point, we should define what is the style convention that we want to use and stick to it.

**Comment by ivan-savciuc** [comment]
Also please squash all the commit's, we have 13 of them and rebase to the latest main to resolve conflicts.

File: flink_job.go (hunk 1 above)
Comment (mhoffm-aiven) [reply flink_job.go:26]: :heavy_check_mark:

File: flink_job.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+			Deploying    int `json:"DEPLOYING"`
+			Failed       int `json:"FAILED"`
+			Finished     int `json:"FINISHED"`
+			Initializing int `json:"INITIALIZING"`
+			Reconciling  int `json:"RECONCILING"`
+			Running      int `json:"RUNNING"`
+			Scheduled    int `json:"SCHEDULED"`
+		} `json:"timestamps"`
+
+		//TODO: if this is needed, add fields
```
[the remark this answers is not in the document]
Comment (mhoffm-aiven) [reply flink_job.go:89]: :heavy_check_mark:

File: flink_job.go (hunk 2 above)
Comment (mhoffm-aiven) [reply flink_job.go:94]: :heavy_check_mark:

File: flink_table.go (hunk 4 above)
Comment (mhoffm-aiven) [reply flink_table.go:43]: :heavy_check_mark:

File: flink_table.go (hunk 5 above)
Comment (mhoffm-aiven) [reply flink_table.go:49]: :heavy_check_mark:

File: flink_table.go (hunk 6 above)
Comment (mhoffm-aiven) [reply flink_table.go:65]: :heavy_check_mark:

File: flink_job.go [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+		JobId string `json:"job_id"`
+	}
+
+	// GetFlinkJobRequest Aiven API request
+	// GET https://api.aiven.io/v1/project/<project>/service/<service_name>/flink/job/<job_id>
+	GetFlinkJobRequest struct {
+		JobId string `json:"job_id"`
+	}
+
+	// GetFlinkJobRequest Aiven API response
```
Comment (ivan-savciuc) [inline flink_job.go:38]: Here should be a GetFlinkJobResponse

**Review by ivan-savciuc — DISMISSED** [review]
Sry, missed this type at the first time, otherwise LGTM

File: flink_job.go (hunk 8 above)
Comment (mhoffm-aiven) [reply flink_job.go]: :heavy_check_mark:

**Review by ivan-savciuc — APPROVED** [review]
LGTM

**Pull request merged**